### PR TITLE
Refactor interface

### PR DESCRIPTION
**Pull request by ilcardella**
I have refactored the NordVPN interface class in order to be independent from the indicator itself. The NordVPN class is merely an set of methods to interact with NordVPN client app. The indicator has its own routine where every X seconds fetches the latest connection status and it updates the menu/icon accordingly.

I have created a few enums and other classes to make it easier (hopefully!) to update/add/remove anything in the future, as it seems they like to change parameters names (see other PR)

Let me know if you see any problem with this.

### CODE REVIEW

**Comment by yorickvanzweeden** [comment]
@ilcardella Due to merging previous pull requests, Github thinks there is a conflict. Please resolve these, as I can't merge it right now

File: code/nordvpn.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
+# NordVPN interface class
+# Provides an interface with the NordVPN Linux client application
+
+import re
+import subprocess
+from enum import Enum
+
+
+class ConnectionStatus(Enum):
```
Comment (yorickvanzweeden) [inline code/nordvpn.py:9]: Please add `@unique annotation`

https://docs.python.org/3/library/enum.html#ensuring-unique-enumeration-values

File: code/nordvpn.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    """
+    Connection status to the VPN
+    """
+    CONNECTED = 'Connected'
+    DISCONNECTED = 'Disconnected'
+    WAITING = 'Connecting'
+
+
+class Settings(Enum):
```
Comment (yorickvanzweeden) [inline code/nordvpn.py:18]: Please add `@unique annotation`

https://docs.python.org/3/library/enum.html#ensuring-unique-enumeration-values

File: code/nordvpn.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    DNS = "DNS"
+    NOTIFY = "Notify"
+
+
+class NordVPNStatus():
+    """
+    Status of the NordVPN client app
+    """
+
+    class Param(Enum):
```
Comment (yorickvanzweeden) [inline code/nordvpn.py:37]: Please add `@unique annotation`

https://docs.python.org/3/library/enum.html#ensuring-unique-enumeration-values

File: code/nordvpn_indicator.py [hunk 4]
```diff
@@ -130,9 +124,9 @@ def build_menu(self):
         item_status.set_submenu(menu_status)
 
         # First item is to refresh the status
-        item_refresh = gtk.MenuItem('Refresh')
-        item_refresh.connect('activate', self.nordvpn.status_check)
-        menu_status.append(item_refresh)
+        # item_refresh = gtk.MenuItem('Refresh')
```
Comment (yorickvanzweeden) [inline code/nordvpn_indicator.py]: Remove unused code?

File: code/nordvpn_indicator.py [hunk 5]
```diff
@@ -332,202 +337,18 @@ def on_setting_update(self, widget):
...
-            if key == NordVPN.Settings.AUTO_CONNECT:
-                if value == 'Off':
-                    setting = False
-                elif value == 'Automatic':
-                    setting = True
-                else:
-                    setting = 'on {}'.format(value.lower())
-            self.run_command('nordvpn set {} {}'.format(key.value.replace(' ','').lower(), str(setting).lower()))
-
+        if widget.get_name() == Settings.PROTOCOL.value:
```
Comment (yorickvanzweeden) [inline code/nordvpn_indicator.py]: I would suggest using a switch case

**Comment by ilcardella** [comment]
I have updated the code with the comments and resolved the conflicts

**Review by yorickvanzweeden — APPROVED** [review]
Thanks for putting so much effort in this repo :)

**Pull request merged**